src/connectors/stats/connectStats.js

```diff
@@ -88,5 +80,24 @@ export default function connectStats(renderFn, unmountFn = noop) {
     dispose() {
       unmountFn();
     },
+
+    getRenderState(renderState, renderOptions) {
+      return {
+        ...renderState,
+        stats: this.getWidgetRenderState(renderOptions),
+      };
+    },
+
+    getWidgetRenderState({ results, helper }) {
+      return {
+        hitsPerPage: results ? results.hitsPerPage : helper.state.hitsPerPage,
```

Comment:
since it's the same condition for all, I think it might be better like this:

```
if (!results) { return { nbHits: 0...}}
return { nbHits: results.nbHits }
```

that will save a bit of computation